### ISSUE TITLE
Auto inflation of JSON data is deprecated.

### ISSUE THREAD

**Issue by daften** [thbishop/knife-solo_data_bag#33]
When creating or editing a data bag with knife solo data bag, I get the following warning:
WARN: Auto inflation of JSON data is deprecated. Please pass in the class to inflate or use #edit_hash at /opt/chefdk/embedded/lib/ruby/2.1.0/forwardable.rb:183:in `edit_data'

**Comment by saidatta** [comment]
same here.

**Comment by saidatta** [comment]
I get this too

**Comment by douglasryanadams** [comment]
Same

**Comment by gabienache** [comment]
Same. Anyone can enlighten us, please?

**Comment by AlexMorreale** [comment]
have been getting this as well

**Comment by mohamed-imran** [comment]
Same here too. Any insights from anyone on it?

**Comment by martijnvanderwoud** [comment]
Same here

**Comment by joeyates** [comment]
I'm unable to reproduce this behavior. I believe it may relate to ChefDK.

Could someone set a breakpoint in `forwardable.rb` to see why/how the method is getting called?

**Comment by AlexMorreale** [comment]
i have to downgrade my chefdk to get this to stop working 

```
halcyon   ➜  chef --version
Chef Development Kit Version: 0.17.17
chef-client version: 12.13.37
delivery version: master (f68e5c5804cd7d8a76c69b926fbb261e1070751b)
berks version: 4.3.3
kitchen version: 1.9.0
```

**Comment by joeyates** [comment]
@daften Could you help reproducing this issue? A backtrace from the point where the warning is given would be very helpful.

**Comment by joeyates** [comment]
OK, so I've found the commit that introduces this warning: https://github.com/chef/chef/commit/8eae0cc429b1ad2f569e611202051a1bddbe142a

**Comment by joeyates** [comment]
The fix is in release 2.1.1

**Comment by daften** [comment]
@joeyates thanks for this. And sorry, but i was swamped with work and while reproducing was on my todo, it got bumped to the bottom because of priorities :s
